### DISCUSSION TITLE
Console return types and App\Console\Kernel::$commands

### DISCUSSION THREAD

**Discussion by petergallagher** [larastan/larastan#2142] [Q&A]
After recently upping the phpstan level on a project I work on we got a lot of errors which should have been resolved by the `ArgumentDynamicReturnTypeExtension` and `OptionDynamicReturnTypeExtension`. On investigation it is because this application has been upgraded through quite a few laravel versions and still has commands registered via the `$commands` array in the console kernel.

If I register the command in a service provider using `$this->commands([CommandName::class])` the extension picks it up and corrects the return types but before I start refactoring how commands are registered I thought I would make sure I'm not missing something that would get it to recognise commands from the kernel and if not was that intentional or just something that didn't happen because it's no longer the recommended way to register commands but could be supported as it does still work?

**Comment by calebdw** [comment]
Here's how I auto register all of our commands:

```php
class Kernel extends ConsoleKernel
{
    protected function commands(): void
    {
        $this->load(array_keys(iterator_to_array(
            Finder::create()
                ->directories()
                ->in([app_path()])
                ->path('Commands'),
        )));
    }
}
```

**Comment by canvural** [comment]
Hi,

I think it is just a missing feature. You can try to send a PR that updates `ConsoleApplicationResolver::findCommands` that will take into account the commands registered via properties.
